spec/unit/counter_spec.rb

```diff
@@ -1,18 +1,37 @@
 require 'counter'
 
 describe Counter do
-  describe '#new' do
-    it 'starts with a count of 0' do
-      counter = Counter.new
-      expect(counter.count).to eq 0
-    end
-  end
+  before (:each) { @counter = Counter.new }
 
   describe '#increment' do
     it 'adds 1 to the count' do
-      counter = Counter.new
-      counter.increment
-      expect(counter.count).to eq 1
+      @counter.increment
+      expect(@counter.count).to eq 1
     end
   end
+
+  describe '#decrement' do
+    it 'takes 1 from the count' do
+      @counter.decrement
+      expect(@counter.count).to eq -1
+    end
+  end
+
+  describe '#reset' do
+    it 'returns the count to 0' do
+      @counter.increment
+      @counter.reset
+      expect(@counter.count).to eq 0
+    end
+  end
+
+  describe '#time' do
+    it 'adds the time of last count' do
+      @counter.increment
+      time = @counter.time
+      expect(@counter.time).to eq time
```

Comment:
This is a vacuous test. It is guaranteed to always pass -- can you see why?

What is the behaviour you really need to test here? Hint: you might have to use mocking or stubbing.